src/Catalog/Registration/RegistrationMakerCatalogItem.cs

```diff
@@ -14,30 +13,37 @@ namespace NuGet.Services.Metadata.Catalog.Registration
 {
     public class RegistrationMakerCatalogItem : CatalogItem
     {
-        Uri _catalogUri;
-        IGraph _catalogItem;
-        Uri _itemAddress;
-        Uri _packageContentBaseAddress;
-        Uri _packageContentAddress;
-        Uri _registrationBaseAddress;
-        Uri _registrationAddress;
-        DateTime _publishedDate;
-        Boolean _listed;
-
-        public RegistrationMakerCatalogItem(Uri catalogUri, IGraph catalogItem, Uri registrationBaseAddress, Uri packageContentBaseAddress = null)
+        private readonly Uri _catalogUri;
+        private readonly IGraph _catalogItem;
+        private Uri _itemAddress;
+        private readonly Uri _packageContentBaseAddress;
+        private Uri _packageContentAddress;
+        private readonly Uri _registrationBaseAddress;
+        private Uri _registrationAddress;
+        private DateTime _publishedDate;
+        private bool _listed;
+        private readonly bool _isExistingItem;
+
+        public RegistrationMakerCatalogItem(Uri catalogUri, IGraph catalogItem, Uri registrationBaseAddress, bool isExistingItem, Uri packageContentBaseAddress = null)
         {
             _catalogUri = catalogUri;
             _catalogItem = catalogItem;
             _packageContentBaseAddress = packageContentBaseAddress;
             _registrationBaseAddress = registrationBaseAddress;
+            _isExistingItem = isExistingItem;
         }
 
         public override StorageContent CreateContent(CatalogContext context)
         {
+            if (_isExistingItem)
+            {
+                return null;
+            }
+
```

Comment:
This behavior is pretty counter intuitive, and makes the code more complex. Is it possible to make a change in a higher layer?